src/server/generic_family.cc

```diff
@@ -1179,13 +1179,15 @@ void GenericFamily::Sort(CmdArgList args, ConnectionContext* cntx) {
     if (bounds) {
       auto sort_it = entries.begin() + std::min(bounds->first + bounds->second, entries.size());
       std::partial_sort(entries.begin(), sort_it, entries.end(),
-                        [reversed](const auto& lhs, const auto& rhs) {
-                          return bool(lhs.Cmp() < rhs.Cmp()) ^ reversed;
+                        [reversed](const auto& lhs, const auto& rhs) -> bool {
+                          return reversed ? rhs.Cmp() < lhs.Cmp() : lhs.Cmp() < rhs.Cmp();
                         });
     } else {
-      std::sort(entries.begin(), entries.end(), [reversed](const auto& lhs, const auto& rhs) {
-        return bool(lhs.Cmp() < rhs.Cmp()) ^ reversed;
-      });
+      std::sort(entries.begin(), entries.end(),
+                [reversed, &entries](const auto& lhs, const auto& rhs) -> bool {
+                  DCHECK(&rhs < &(*entries.end()) && &rhs >= &(*entries.begin()));
```

Comment:
yeah but the implementation is correct because it's vector. it's under DCHECK anyways.

Reply:
what do you mean by the implementation is correct ? If you mean that it's safe to dereference this then it's not. https://en.cppreference.com/w/cpp/container/vector/end and https://eel.is/c++draft/iterator.requirements.general#7 

> This element acts as a placeholder; attempting to access it results in undefined behavior.

Now if you mean the implementation (the standard library provided by the system) then yeah sure it might work but I would not rely on it at any point.

Also since `entries` is a vector it's actually less verbose to turn this check into `<= &entries.back()` and the comperator would never be used by sort with and end() iterator. 

Anyway it's a DCHECK and since it's now causing any issues :shrug:

Reply:
oh you changed it :heart:

Reply:
But I still disagree. There is no asm code that reads from that address. I request `&(*it)` and yes, if I would **read** `*it` it would be UB but I do not. So the standard can tell me many things but at the end, the code won't segfault on any compiler.

Reply:
I see, thank you for explaining this